Code/Tools/FBuild/FBuildCore/Protocol/Client.cpp

```diff
@@ -762,14 +1175,69 @@ bool Client::WriteFileToDisk( const AString & fileName, const char * data, const
 
 // CONSTRUCTOR( ServerState )
 //------------------------------------------------------------------------------
-Client::ServerState::ServerState()
+Client::ServerState::ServerState( const bool detailedLogging )
     : m_Connection( nullptr )
     , m_CurrentMessage( nullptr )
-    , m_NumJobsAvailable( 0 )
+    , m_ConnectionRetryIntervalSec( 0 )  // try immediately the first time
+    , m_ConnectionRetryElapsedTimeSec( 0 )
+    , m_NumJobsAvailableForWorker( 0 )
     , m_Jobs( 16, true )
     , m_Blacklisted( false )
+    , m_Excluded( false )
+    , m_DetailedLogging( detailedLogging )
+{
+}
+
+// DESTRUCTOR( ServerState )
+//------------------------------------------------------------------------------
+Client::ServerState::~ServerState()
+{
+    MutexHolder mh( m_Mutex );
```

Comment:
holding this mutex in the destructor is suspect. It implies that something else could be trying to access this class while it is being destructed, which is invalid.